### PR TITLE
Add fetch api title on click

### LINKED ISSUE #2: Fetch post api

### PR DESCRIPTION
**Pull request by Kunal1101**
image

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            
+        });
+    } catch (error) {
+        console.error('There was a problem with the fetch operation:', error);
+    }
+}
+
+
+const button = document.getElementById('fetch-api-button');
+button.addEventListener('click', () => fetchTodo())
```
Comment (satyam-seth) [inline script.js]: Add event listener on window load

**Review by satyam-seth — CHANGES_REQUESTED** [review]

File: script.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+async function fetchTodo() {
```
Comment (satyam-seth) [inline script.js]: `fetchTodo` function only responsible for fetching data from the API 

create another method to append skeleton for data in the dom

**Review by satyam-seth — CHANGES_REQUESTED** [review]

File: script.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+const appendData = (title) => {
+    const container = document.querySelector('.append-titles')
+    const newTitle = document.createElement('span')
+    newTitle.innerHTML =  title;
+    container.appendChild(newTitle)
+}
+
+window.onload = () => {
+    const button = document.getElementById('fetch-api-button');
+    button.addEventListener('click', () => fetchTodo())
```
Comment (satyam-seth) [inline script.js]: no need to create another anonymous function callback for click listener
wdyt?

```suggestion
    button.addEventListener('click', fetchTodo)
```

File: script.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+async function fetchTodo() {
+    try {
+        const response = await fetch('https://jsonplaceholder.typicode.com/todos');
+        if (!response.ok) {
+            throw new Error('Network response was not ok');
+        }
+        const json = await response.json();
+        json.forEach((item) => appendData(item.title));
```
Comment (satyam-seth) [inline script.js]: `fetchTodo` is only responsible for fetching data.

`appendData` should call within window.onload

**Review by satyam-seth — CHANGES_REQUESTED** [review]

File: script.js (hunk 3 above)
Comment (Kunal1101) [reply script.js]: fixed

File: script.js (hunk 4 above)
Comment (Kunal1101) [reply script.js]: fixed

File: script.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        }
+        const json = await response.json();
+        return json
+    } catch (error) {
+        console.error('There was a problem with the fetch operation:', error);
+    }
+}
+
+ async function appendData() {
+    const data =  await fetchTodo();
```
Comment (satyam-seth) [inline script.js]: `appendData` will take data as a parameter and perform its task; it is not responsible for fetching the data.

**Review by satyam-seth — CHANGES_REQUESTED** [review]

File: script.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        const json = await response.json();
+        return json;
+    } catch (error) {
+        console.error('There was a problem with the fetch operation:', error);
+    }
+}
+
+
+// Append data in DOM 
+async function appendData(data) {
```
Comment (satyam-seth) [inline script.js]: Why is this function asynchronous?

Comment (Kunal1101) [reply script.js]: I thought this will wait to load data from fetch, that's why

**Review by satyam-seth — APPROVED** [review]

**Pull request merged**